frontend/src/components/shared/Modal/Modal.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { ReactNode, ReactElement } from "react"
+import {
+  Modal,
+  ModalHeader as UIModalHeader,
+  ModalBody as UIModalBody,
+  ModalFooter,
+  ModalButton as UIModalButton,
+} from "baseui/modal"
+import { KIND, ButtonOverrides } from "baseui/button"
+import { Kind } from "components/shared/Button"
```

Comment:
See other comment. This is a temporary measure until the migration to Button. Essentially Because Modal exports a `ModalButton`, I opted to start a `Button` component migration but save it for a different PR. The logic of deciding how the button should look should be in that `Button` component. The benefit of this change now is I won't have to change the kinds on the use of `ModalButton`. (ie the interface will remain the same, but the implementation will be changed underneath)